metadata/src/test/java/org/apache/kafka/metadata/migration/KRaftMigrationDriverTest.java

```diff
@@ -627,10 +629,13 @@ public void testTopicDualWriteDelta() throws Exception {
 
             assertEquals(1, topicClient.deletedTopics.size());
             assertEquals("foo", topicClient.deletedTopics.get(0));
-            assertEquals(1, topicClient.createdTopics.size());
-            assertEquals("baz", topicClient.createdTopics.get(0));
+            assertEquals(2, topicClient.createdTopics.size());
+            assertTrue(topicClient.createdTopics.contains("foo"));
+            assertTrue(topicClient.createdTopics.contains("baz"));
             assertTrue(topicClient.updatedTopicPartitions.get("bar").contains(0));
-            assertEquals(new ConfigResource(ConfigResource.Type.TOPIC, "foo"), configClient.deletedResources.get(0));
+            assertEquals(2, configClient.deletedResources.size());
```

Comment:
It's strange that `configClient` tracks two deleted resources while `topicClient` only sees one. 
The only difference between this and the former test seems to be that the migration driver calls `onMetadataUpdate` on a `LogDeltaManifest` here rather than a `SnapshotManifest` in the former test.
cc @mumrah